lib/ui/plugins.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of dart.ui;
+
+/// An wrapper for a raw callback handle.
+class CallbackHandle {
+  final int _handle;
+
+  /// Create an instance using a raw callback handle.
+  ///
+  /// Only values produced by a call to [CallbackHandle.toRawHandle] should be
+  /// used, otherwise this object will be an invalid handle.
+  CallbackHandle.fromRawHandle(this._handle)
+      : assert(_handle != null, "'_handle' must not be null.");
+
+  /// Get the raw callback handle to pass over a [MethodChannel] or isolate
+  /// port.
+  int toRawHandle() => _handle;
+
+  @override
+  int get hashCode => _handle;
+
+  @override
+  bool operator ==(dynamic other) =>
+      (other is CallbackHandle) && (_handle == other._handle);
+}
+
+/// Functionality for Flutter plugin authors.
+abstract class PluginUtilities {
+  /// Get a handle to a named top-level or static callback function which can
+  /// be easily passed between isolates.
+  ///
+  /// `callback` must not be null.
+  ///
+  /// Returns a [CallbackHandle] that can be provided to
+  /// [PluginUtilities.getCallbackFromHandle] to retrieve a tear-off of the
+  /// original callback. If `callback` is not a top-level or static function,
+  /// null is returned.
+  ///
+  /// The result of this method should be cached in order to avoid repeated
```

Comment:
Oh, I guess you could add a cache in this class.
static Map<Function, int> _forwardCache;
static Map<int, Function> _backwardCache;

getCallbackHandle(Function callback) {
  // return if it's in _forwardCache
  // otherwise do the _getCallbackHandle and add to both _forwardCache and _backwardCache
}

and similar for getCallbackFromHandle.